test/core/end2end/cq_verifier.c

```diff
@@ -71,17 +70,16 @@ typedef struct expectation {
 struct cq_verifier {
   /* bound completion queue */
   grpc_completion_queue *cq;
-  /* the root/sentinal expectation */
-  expectation expect;
+  /* expectation list */
+  expectation *first_expectation;
+  expectation *last_expectation;
```

Comment:
When adding new expectations to the list, we always add them at the end.  Having a pointer to the last element of the list makes this a bit more efficient, since otherwise we'd have to iterate through the list to find the last element.

I suppose efficiency doesn't really matter here, since (a) this is just test code, and (b) I suspect we don't have any tests where the number of expectations is large.  So if you'd prefer, I can eliminate last_expectation and change the add() function to iterate to the end of the list.